### DISCUSSION TITLE
How to get Current user in hook model ? Adonis v 4.1

### DISCUSSION THREAD

**Discussion by miqbalrr** [adonisjs/core#1575] [Help]
I'd like to create a new event that will delete cache after data saved into database. 
i have caching system use redis, from another app in adonis with key as logged user. 

so i need the user id from model to delete cache by specifiec key.

`this.addHook('afterSave', async() => { `
`            const data = {key: "prefix_userID"}`
`            Bull.add(ClearCacheJobs.key, data )`
`})`

thankyou.

**Comment by mathieuh** [comment]
Hi @miqbalrr,

If you're in your User model you can simply do this :
`this.addHook('afterSave', async (user) => { })`

It's documented here: https://adonisjs.com/docs/4.1/database-hooks#_binding_closure

Cheers

**Reply by miqbalrr** [reply]
Sorry for my bad question, what i mean is, i want to get a current logged user from auth.user.id inside model

**Reply by mathieuh** [reply]
Ok I don't think it is possible to access the current logged user inside a model.

Could you show the model and the code that triggers the `save()`?
Maybe you could delete the cache from outside the model.

**Reply by miqbalrr** [reply]
i have made my own helper for dispatching job through controllers, yeah i don't think it is possible to get logged user in model.
this the helpers.

````
'use strict'

const Bull = use('Rocketseat/Bull')
const { ioc } = require('@adonisjs/fold')
const instance = "App/Jobs/"

class Dispatch {
    constructor(job, param) {
        this._instance = instance + job
        this._parameter = param
    }

    async now() {
        const Job = ioc.use(this._instance)
        Bull.add(Job.key, this._parameter)
    }

}

module.exports = Dispatch
````
so i just called 
````
new Dispatch('jobname', 'param with logged user').now()
````
